src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -124,7 +124,15 @@ private void OnWorkspacePropertyChanged(object sender, PropertyChangedEventArgs
         internal void DependencyRegen(WorkspaceModel ws)
         {
             this.RestartBanner.Visibility = Visibility.Hidden;
-            var packageDependencies = ws.NodeLibraryDependencies.Where(d => d is PackageDependencyInfo);
+            var packageDependencies = ws.NodeLibraryDependencies.Where(d => d is PackageDependencyInfo).ToList();
+
+            // This assumes it is a one-time addition to add the IronPython extension 
+            // as a package dependency. We might only want to give users the option to install
+            // the package on opening a workspace for the first-time. Even if any IronPython nodes are deleted
+            // from the workspace subsequently, it should not be necessary to uninstall/reinstall the package again later.
```

Comment:
@aparajit-pratap Can you talk about it more? Let's say if the user was hesitate to address it first time opening, then the chance is gone?

Reply:
If the user doesn't download it the first time, it will continue to display as a missing dependency in the workspace references view, just like any other missing package.

Reply:
I should make the comment clearer. What I meant to say is that if a python node is removed from the workspace, the workspace references may not be updated. I need to double-check.